### PR TITLE
ZIOS-9804: Added back button in the email verification view

### PR DESCRIPTION
**Pull request by nicorsm**
## What's new in this PR?

Added back button on the email verification view, so now users can go back to the previous view when they provide a wrong email address. Added tests.

### CODE REVIEW

**Comment by vytis** [comment]
I think this is the step where we keep trying to login with this email every 5 mins. We need to stop this process somehow, but probably there is a method to reset the authentication state

File: Wire-iOS/WireBridgingHeader.h [hunk 1]
```diff
@@ -96,6 +96,7 @@
 #import "ConfirmAssetViewController.h"
 #import "ProfileSelfPictureViewController.h"
 #import "AddEmailPasswordViewController.h"
+#import "EmailVerificationStepViewController.h"
```
Comment (mikeger) [inline Wire-iOS/WireBridgingHeader.h]: You sure this is necessary?

Comment (nicorsm) [reply Wire-iOS/WireBridgingHeader.h]: I need it just for test purposes, will move to the proper tests bridging header.

**Comment by vytis** [comment]
I think stopping the timer should be possible by resetting the authentication status somehow - clearing credentials or firing a notification. You should check if there is a situation already where we quit the authentication flow an see what is done there.

**Comment by nicorsm** [comment]
@vytis just found that we're already calling `[[UnauthenticatedSession sharedSession] cancelWaitForEmailVerification]` in `viewWillDisappear`. Need to check the appearing/disappearing of the back button since it's not working on the registration flow.

**Pull request merged**